index.bs

```diff
@@ -2943,6 +3027,53 @@ and an optional [=boolean=] |isNoised| (default false):
 1. Run [=obtain and deliver an aggregatable debug report on source registration=]
     with |dataType|, |source|, and |isNoised|.
 
+To <dfn>find sources with common destinations and reporting origin</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |matchingSources| be a new [=list=].
+1. [=set/iterate|For each=] |source| of the user agent's [=attribution source cache=]:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. [=list/Append=] |source| to |matchingSources|.
+1. Return |matchingSources|.
+
+To <dfn>check if source has incompatible attribution scope fields</dfn> given an [=attribution source=] |source| and an [=attribution source=] |pendingSource|:
+1. If |pendingSource|'s [=attribution source/max event states=] is not null and |source|'s [=attribution source/max event states=] is not null and does not equal |pendingSource|'s [=attribution source/max event states=], return true.
+1. If |pendingSource|'s [=attribution source/attribution scope limit=] is not null and |source|'s [=attribution source/attribution scope limit=] is not null and less than |pendingSource|'s [=attribution source/attribution scope limit=], return true.
+1. Return false.
+
+To <dfn>remove sources with incompatible attribution scope fields from cache</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |matchingSources| be the result of running [=find sources with common destinations and reporting origin=] with |pendingSource|.
+1. [=list/iterate|For each=] |source| of |matchingSources|:
+    1. If the result of running [=check if source has incompatible attribution scope fields=] with |source| and |pendingSource| is true:
+        1. If |source|'s [=attribution source/randomized response=] is not null:
```

Comment:
What happens if there's a real report whose trigger time is later? Shall we delete that report and adjusts the attribution limit?

Reply:
The rationale to delete pending reports which have trigger_time > new source’s registration is to hide the fact that source was noised. I think the assumption we made was that only fake reports can have a future trigger time, I think we shouldn't differentiate fake / real event reports here. i.e. they should be both sent or neither should be sent.

@apasel422 wdyt?

Reply:
A real report can have a future trigger time because [the wall time](https://w3c.github.io/hr-time/#dfn-current-wall-time) is not necessarily monotonically increasing. This is actually an existing problem in the spec unrelated to this PR, but if possible we should avoid adding even more assumptions based on it.

Reply:
Per discussion offline,  from privacy perspective, we shouldn't treat fake/true reports differently here. agreed that we shouldn't rely on `walltime` to determine whether it's a fake report or not.